### ISSUE TITLE
Docker directory mapping problems

### ISSUE THREAD

**Issue by suckerfish** [prologic/tube#18]
Instructions for docker say to map '/data' within the container to wherever.

However, nothing is written to '/data' within the container after an upload. All uploads seem to live within '/go' and then a couple subfolders.  But if I try to map '/go' to my host system, I get errors about renaming files and something about a $PATH statement being wrong. 

So until the mappings can get solved, it is writing videos into the container which is not desirable.

**Comment by prologic** [comment]
Thanks for reporting. Yes the docs are slightly wrong on this. I will fix this. PR is welcome however as I have quite a few issues to fix :)

**Comment by suckerfish** [comment]
Sorry I'm pretty new to github and not a coder. I would like to help. If it's just docs that are wrong, can you point me in the direction of how to find out where the correct info is? Maybe I can then figure out how to do a PR correctly.

**Comment by prologic** [comment]
If you to the repo itself and viae the README there's a pencil icon you can use to edit the docs in-place and submit a PR :)

As for the "correct" info its somewhat in the `docker-compose.yml` as I run it this way in production myself using Docker in a Swarm cluster.

I'll try to get the docs fixed/updated if you don't beat me to it :)

**Comment by suckerfish** [comment]
Yeah I saw the docker-compose.yml and the heart of the matter is that _as far as I can tell_ (I'm not a super expert with Docker or compose) it's trying to map the container directory '/data' to somewhere else.

I do this all the time with my other containers. Map internal directory 'config' to somewhere on the host system, map a path to storage within the container to where those files are supposed to live on your host system. I do this with Plex, Nextcloud, and countless others. But the app needs to actually store things in that directory in the container. So if I'm mapping /data to a directory on my host system, the app should be actually writing things to /data or what is the point of mapping it. Right? 

So what exactly is Tube supposed to be writing to '/data' within the container? Because as far as I can tell, it's not writing anything there. 

And to reiterate, it might just be my ignorance or something I'm missing, but just to be clear I'm trying to map a directory to a storage location on my host system where all the uploaded videos will be kept.

**Comment by prologic** [comment]
This still needs to be solved for users that run this in Docker besides myself at tube.mills.io -- I would encourage the community to pick this up if you can as I have very little bandwidth due to my day job at the moment. Thanks!

**Comment by fxjkhr** [comment]
from what I can tell, the "tube" docker container looks for videos in `/go/videos` so... `$ docker run -p 8000:8000 -v /videos:/go/videos prologic/tube` should work

**Comment by prologic** [comment]
This can probably be improved somewhat by improving the `Dockerfile` itself a bit.

**Comment by Skayo** [comment]
Simplest solution is to configure all the paths to point to `/data`:

```json
{
    "library": [
        {
                "path": "/data/videos",
                "prefix": ""
        }
    ],
    "server": {
        "store_path": "/data/tube.db",
        "upload_path": "/data/uploads"
    }
}
```
